driver/bpf/filler_helpers.h

```diff
@@ -892,14 +926,33 @@ static __always_inline int __bpf_val_to_ring(struct filler_data *data,
 
 #ifdef BPF_FORBIDS_ZERO_ACCESS
 				if (read_size)
-					if (bpf_probe_read(&data->buf[curoff_bounded],
-							   ((read_size - 1) & SCRATCH_SIZE_HALF) + 1,
-							   (void *)val))
+					if (mem == KERNEL)
+					{
+						rc = bpf_probe_read_kernel(&data->buf[curoff_bounded],
+								  ((read_size - 1) & SCRATCH_SIZE_HALF) + 1,
+								  (void *)val);
+					}
+					else
+					{
+						rc = bpf_probe_read_user(&data->buf[curoff_bounded],
+								  ((read_size - 1) & SCRATCH_SIZE_HALF) + 1,
+								  (void *)val);
+					}
```

Comment:
Same as my 2 previous comments, put curly braces around the block or move it into its own macro/function.